packages/datadog-instrumentations/src/apollo-server-core.js

```diff
@@ -16,30 +16,21 @@ addHook({ name: 'apollo-server-core', file: 'dist/runHttpQuery.js', versions: ['
         return originalRunHttpQuery.apply(this, arguments)
       }
 
-      const asyncResource = new AsyncResource('bound-anonymous-fn')
-      return asyncResource.runInAsyncScope(async () => {
-        startApolloServerCoreRequest.publish()
+      startApolloServerCoreRequest.publish()
 
-        const runHttpQueryResult = await originalRunHttpQuery.apply(this, arguments)
+      const runHttpQueryResult = await originalRunHttpQuery.apply(this, arguments)
 
-        return asyncResource.runInAsyncScope(() => {
-          const abortController = new AbortController()
-          const abortData = {}
-          successApolloServerCoreRequest.publish({ abortController, abortData })
+      const abortController = new AbortController()
+      const abortData = {}
+      successApolloServerCoreRequest.publish({ abortController, abortData })
```

Comment:
This could also be done with tracing channel by using tracePromise and constructing the abort controller earlier. Could probably also make it a `Promise.race(...)` between the `originalRunHttpQuery` call and a promise reject from the abort.

Reply:
Good suggestion, a bit cleaner this way.